DEVELOPER.md

```diff
@@ -0,0 +1,43 @@
+# Notes for Development
+
+## Development Environment
+
+## Building from Source
+
+To build and test the driver, the standard cargo commands can be used from the root directory.
+
+For an unoptimized build with debugging information (most common):
+```
+cargo build
+```
+
+For an optimized build with debugging info (mac)
+```
+cargo build --features odbc-sys/iodbc,cstr/utf32 --profile=release-with-debug
+```
+And for an optimized build with debugging info (windows, linux)
+```
+cargo build --profile=release-with-debug
+```
+The resulting build files will be output to the `target` directory in the `debug` or `release` folders respectively.
+
+## Running Tests
+
+### To run unit tests
+
+Similar to building, standard cargo commands can be used here:
+
+```
+cargo test unit
```

Comment:
The tests should also be run with `--features` needed

Reply:
👍 just to double check, does this 1. only apply to mac? and 2. does it apply to unit tests or just to the build (which I should update), just looking at evergreen file I see the features in `compile macos <debug/release>` but not `run macos unit tests`

Reply:
I believe it only applies to mac. As a general rule, it should be done for all tests. Ones that may be impacted that come to mind are the data tests which may very much depend on the length of the buffer returned.